lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1592,6 +1660,61 @@ def request_certificate_renewal(
         )
         logger.info("Certificate renewal request completed.")
 
+    def get_assigned_certificates(self) -> List[Dict[str, str]]:
+        """Gets a list of certificates that were assigned to this unit.
+
+        Returns:
+            List of certificates. For example:
+            [
+                {
+                    "ca": "-----BEGIN CERTIFICATE-----...",
+                    "chain": [
+                        "-----BEGIN CERTIFICATE-----..."
+                    ],
+                    "certificate": "-----BEGIN CERTIFICATE-----...",
+                    "certificate_signing_request": "-----BEGIN CERTIFICATE REQUEST-----...",
+                }
+            ]
+        """
+        final_list = []
+        for csr in self.get_certificate_signing_requests(fulfilled_only=True):
+            assert type(csr["certificate_signing_request"]) == str
+            if cert := self._find_certificate_in_relation_data(csr["certificate_signing_request"]):
+                final_list.append(cert)
+        return final_list
+
+    def get_certificate_signing_requests(
+        # RFC: Should I just split this up into 3 different functions?
```

Comment:
If we have independent usecases for those 3 circumstances, I agree that 3 different methods makes sense. It's generally preferable to have smaller functions that have 1 purpose instead of multiple.

Let's only create the methods we know we need though. For example if we know we need 2 out of those 3, let's not implement the third one "just in case".

Reply:
I don't have a very strong opinion on this, I'll let you make the call @Kayra1 as you're right in it at the moment.

Reply:
This is fine